app/Http/Controllers/TodoController.php

```diff
@@ -39,16 +40,16 @@ public function show(Todo $todo): TodoResource
         return new TodoResource($todo);
     }
 
-    public function update(UpdateTodoRequest $request, Todo $todo, UpdateTodoAction $updateTodoAction): TodoResource
+    public function update(UpdateTodoRequest $request, int $id, UpdateTodoAction $updateTodoAction): TodoResource
     {
         $data = new UpdateTodoDTO($request->validated());
-        $updateTodoAction->execute($todo, $data);
+        $todo = $updateTodoAction->execute($id, $data);
         return new TodoResource($todo);
     }
 
-    public function destroy(Todo $todo, DestroyTodoAction $destroyTodoAction): TodoResource
+    public function destroy(int $id, DestroyTodoAction $destroyTodoAction): TodoResource
```

Comment:
Why are you returning the model when you just deleted it? You just need to return 200 status and an empty response